examples/iql/iql_online.py

```diff
@@ -2,291 +2,138 @@
-
-    # Make Off-Policy Collector
-    collector = SyncDataCollector(
-        env_factory,
-        create_env_kwargs={"num_workers": cfg.env_per_collector},
-        policy=model[0],
-        frames_per_batch=cfg.frames_per_batch,
-        max_frames_per_traj=cfg.max_frames_per_traj,
-        total_frames=cfg.total_frames,
-        device=cfg.collector_device,
-    )
-    collector.set_seed(cfg.seed)
+    # Make Loss
+    loss_module, target_net_updater = make_loss(cfg.loss, model)
 
-    # Make Replay Buffer
-    replay_buffer = make_replay_buffer(
-        buffer_size=cfg.buffer_size, device=device, batch_size=cfg.batch_size
-    )
-
-    # Optimizers
-    params = list(loss_module.parameters())
-    optimizer = optim.Adam(params, lr=cfg.lr, weight_decay=cfg.weight_decay)
+    # Make Optimizer
+    optimizer = make_iql_optimizer(cfg.optim, loss_module)
 
     rewards = []
     rewards_eval = []
-
     # Main loop
     collected_frames = 0
-
-    pbar = tqdm.tqdm(total=cfg.total_frames)
+    pbar = tqdm.tqdm(total=cfg.collector.total_frames)
     r0 = None
-    loss = None
+    q_loss = None
 
-    for i, tensordict in enumerate(collector):
+    init_random_frames = cfg.collector.init_random_frames
+    num_updates = int(
+        cfg.collector.env_per_collector
+        * cfg.collector.frames_per_batch
+        * cfg.optim.utd_ratio
+    )
+    prb = cfg.replay_buffer.prb
+    env_per_collector = cfg.collector.env_per_collector
+    eval_iter = cfg.logger.eval_iter
+    frames_per_batch, frame_skip = cfg.collector.frames_per_batch, cfg.env.frame_skip
+    eval_rollout_steps = cfg.collector.max_frames_per_traj // frame_skip
 
+    for i, tensordict in enumerate(collector):
         # update weights of the inference policy
         collector.update_policy_weights_()
 
         if r0 is None:
             r0 = tensordict["next", "reward"].sum(-1).mean().item()
         pbar.update(tensordict.numel())
 
-        if "mask" in tensordict.keys():
-            # if multi-step, a mask is present to help filter padded values
-            current_frames = tensordict["mask"].sum()
-            tensordict = tensordict[tensordict.get("mask").squeeze(-1)]
-        else:
-            tensordict = tensordict.view(-1)
-            current_frames = tensordict.numel()
+        tensordict = tensordict.view(-1)
+        current_frames = tensordict.numel()
         replay_buffer.extend(tensordict.cpu())
         collected_frames += current_frames
 
-        (
-            actor_losses,
-            q_losses,
-            value_losses,
-        ) = ([], [], [])
         # optimization steps
-        for _ in range(cfg.frames_per_batch * int(cfg.utd_ratio)):
-            # sample from replay buffer
-            sampled_tensordict = replay_buffer.sample(cfg.batch_size).clone()
-
-            loss_td = loss_module(sampled_tensordict)
+        if collected_frames >= init_random_frames:
+            (actor_losses, q_losses, value_losses) = ([], [], [])
```

Comment:
One way to do this is
```python
td_log = TensorDict({}, [num_updates])
for i in range(num_updates):
    loss_td = loss_module(sampled_tensordict)
    td_log[i] = loss_td.detach()
    ....
td_log = td_log.apply(lambda x: x.float().mean(), batch_size=[])

```
and then you can just log each value in that tensordict in wandb